homeassistant/components/smlight/config_flow.py

```diff
@@ -97,8 +97,13 @@ async def async_step_zeroconf(
         mac = discovery_info.properties.get("mac")
         # fallback for legacy firmware
         if mac is None:
-            info = await self.client.get_info()
-            mac = info.MAC
+            try:
+                info = await self.client.get_info()
+                mac = info.MAC
```

Comment:
only have code in the try block that can raise, it looks like the `info.MAC` is only fetching a property and thus doesn't raise

Reply:
yeh its `get_info()` that raise the error.